lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -89,7 +89,7 @@ def execute(powershell_code, timeout_ms = nil, working_dir = nil)
             (prop.text.nil? ? nil : Base64.decode64(prop.text))
           # if err contains data it must be "real" stderr output
           # which should be appended to what PS has already captured
-          value += err if err && (err != []) && (name == 'stderr')
+          value = (value || '') + err.join('') if err && (err != []) && (name == 'stderr')
```

Comment:
If `prop.text.nil?`, value remains nil from lines 87-89.
If `err` is nil or the name doesn't equal stderr.